packages/features/bookings/Booker/Booker.tsx

```diff
@@ -0,0 +1,196 @@
+import type { MotionStyle } from "framer-motion";
+import { LazyMotion, domAnimation, m, AnimatePresence } from "framer-motion";
+import { Fragment, useEffect } from "react";
+import StickyBox from "react-sticky-box";
+import { shallow } from "zustand/shallow";
+
+import type { Dayjs } from "@calcom/dayjs";
+import CustomBranding from "@calcom/lib/CustomBranding";
+import classNames from "@calcom/lib/classNames";
+import useMediaQuery from "@calcom/lib/hooks/useMediaQuery";
+import { ToggleGroup } from "@calcom/ui";
+import { FiCalendar, FiColumns, FiGrid } from "@calcom/ui/components/icon";
+
+import { AvailableTimeSlots } from "./components/AvailableTimeSlots";
+import { Away } from "./components/Away";
+import { BookEventForm } from "./components/BookEventForm";
+import { DatePicker } from "./components/DatePicker";
+import { EventMeta } from "./components/EventMeta";
+import { LargeCalendar } from "./components/LargeCalendar";
+import { BookerSection } from "./components/Section";
+import { fadeInUp, fadeInLeft, resizeAnimationConfig } from "./config";
+import { useBookerStore, useInitializeBookerStore } from "./store";
+import type { BookerLayout, BookerProps } from "./types";
+import { useGetBrowsingMonthStart } from "./utils/dates";
+import { useEvent } from "./utils/event";
+
+// @TODO: Test embed view
+/* @TODO: eth signature / gates */
+
+const BookerComponent = ({ username, eventSlug, month, rescheduleBooking }: BookerProps) => {
+  const [browsingMonthStart, setBrowsingMonthStart] = useGetBrowsingMonthStart(month);
+  // Custom breakpoint to make calendar fit.
+  const isMobile = useMediaQuery("(max-width: 800px)");
+  const StickyOnDesktop = isMobile ? Fragment : StickyBox;
+  const rescheduleUid =
+    typeof window !== "undefined" ? new URLSearchParams(window.location.search).get("rescheduleUid") : null;
+  const event = useEvent();
+  const [layout, setLayout] = useBookerStore((state) => [state.layout, state.setLayout], shallow);
+  const [bookerState, setBookerState] = useBookerStore((state) => [state.state, state.setState], shallow);
+  const [selectedDate, setSelectedDate] = useBookerStore(
+    (state) => [state.selectedDate, state.setSelectedDate],
+    shallow
+  );
+  const [selectedTimeslot, setSelectedTimeslot] = useBookerStore(
+    (state) => [state.selectedTimeslot, state.setSelectedTimeslot],
+    shallow
+  );
+
+  useInitializeBookerStore({
```

Comment:
By storing the most important data of the booker in a Zustand store, we have easy access to all this data in child components, without prop drilling.